src/views/List.jsx

```diff
@@ -16,28 +16,30 @@ export function List({ data, listToken }) {
 
 	return (
 		<>
-      <h2>Smart Shopping List</h2>
+			<h2>Smart Shopping List</h2>
 			{listToken && (
 				<p>Want to share your list? Your list token is: {listToken}</p>
 			)}
-			<form onSubmit={submitHandler}>
-				<label htmlFor="itemInput">Filter Items: </label>
-				<input
-					type="text"
-					id="itemInput"
-					name="itemInput"
-					placeholder="Start typing here..."
-					value={listFilter}
-					onChange={(event) => setListFilter(event.target.value)}
-				/>
-				{listFilter && (
-					<span>
-						<button type="reset" onClick={clickHandler}>
-							X
-						</button>
-					</span>
-				)}
-			</form>
+			{data.length > 0 ? (
+				<form onSubmit={submitHandler}>
+					<label htmlFor="itemInput">Filter Items: </label>
+					<input
+						type="text"
+						id="itemInput"
+						name="itemInput"
+						placeholder="Start typing here..."
+						value={listFilter}
+						onChange={(event) => setListFilter(event.target.value)}
+					/>
+					{listFilter && (
+						<span>
+							<button type="reset" onClick={clickHandler}>
+								X
+							</button>
+						</span>
+					)}
+				</form>
+			) : null}
```

Comment:
I think this will work fine as-is, but might be a little more readable if it was just
`data.length > 0 && <form stuff here> `
rather than having a ternary statement with null.

Reply:
@a-fink I actually switched back to ternaries for some of these cases after reading this article: https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Reply:
Thanks for sharing that @mentalcaries ! I wasn't aware of that issue.

Reply:
@mentalcaries thanks for sharing, that makes sense. Was aware of 0 being falsey in JS, but wasn't thinking of how JS/React would interpret it in the `0 && something` case. Very cool to know!

Reply:
nice catch! We do the same at my work, there's an eslint rule for it called [react/jsx-no-leaked-render](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md)

Reply:
> @a-fink I actually switched back to ternaries for some of these cases after reading this article: https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Great article and awesome to know this is good practice.